src/routes/donations.js

```diff
@@ -325,11 +325,18 @@ function _validateIncomingDonation(incoming) {
  * they are marked as a *new* donor.
  */
 
+function uniqByReceiptNo(array){
```

Comment:
I think this function is fine.. but since we are already using lodash's `uniqBy`, you could also use that.